### PR TITLE
Widget Editor: Don't disable the Save button

### PR DESCRIPTION
**Pull request by t-hamano**
Similar to #42842

## What?

PR removes the `disabled` attribute from the Widget Editor's save button component.

widget-editor-save-button

## Why?

#42802 about the Site editor explains why having both disabled and aria-disabled is a problem.

## How?

Moved the variable that defines whether the `disabled` attribute is enabled to the `aria-disabled` attribute and removed the `disabled` attribute.

## Testing Instructions

- Activate TT1 theme.
- Access Appearance > Widgets.
- Open the browser's developer tools.
- The save button should not have `disabled` attribute and `aria-disabled` should be `true`.
- Make changes to the content.
- The save button should not have `disabled` attribute and `aria-disabled` should be `false`.
- Click the save button.
- The save button should not have `disabled` attribute and `aria-disabled` should be `true`.

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/54422211/1459788c-2394-4c5a-88a5-1029d4ffe620

### CODE REVIEW

File: packages/edit-widgets/src/components/save-button/index.js [hunk 1]
```diff
@@ -26,9 +26,8 @@ function SaveButton() {
 		<Button
 			variant="primary"
 			isBusy={ isSaving }
-			aria-disabled={ isSaving }
+			aria-disabled={ isSaving || ! hasEditedWidgetAreaIds }
 			onClick={ isSaving ? undefined : saveEditedWidgetAreas }
```
Comment (Mamaduka) [inline packages/edit-widgets/src/components/save-button/index.js]: I think the condition here also needs to be updated. A user should not be able to click on the button when `! hasEditedWidgetAreaIds`.

PS. We should probably define `isDisabled` condition.

Comment (t-hamano) [reply packages/edit-widgets/src/components/save-button/index.js]: Thanks for the review! I updated via 16282ad, does this make sense?

**Review by draganescu — APPROVED** [review]
Thank you @t-hamano and @Mamaduka - this makes sense to me so I'll ✅ it.

**Comment by t-hamano** [comment]
@Mamaduka @draganescu Thanks for the review!

It seems that React Native E2E Tests are also failing on trunk, so I would like to update and merge this PR after that issue is resolved.

**Pull request merged**